server: improve TestTenantInterface

yuzefovich opened the pull request:
Addresses: #76378.
Fixes: #106903.

yuzefovich committed:
server: return test tenant's things right away

This commit makes it so that if we started the default test tenant, then
we use that tenant's objects for the following methods:
- `ClusterSettings`
- `DiagnosticsReporter`
- `ExecutorConfig` (and `Codec` implicitly since it uses `ExecutorConfig`)
- `JobsRegistry`
- `PGPreServer` and `SQLServer`
- `TenantStatusServer`
- `Tracer` and `TracerI`.

Release note: None

yuzefovich committed:
WIP on removing no longer needed TenantOrServer calls

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/106968)
<!-- Reviewable:end -->

yuzefovich committed:
server: expose system tenant's ExecutorConfig separately

The following commit will make it so that existing
`TestTenantInterface.ExecutorConfig` method returns the tenant's
ExecutorConfig when the test tenant is started, but some tests require
access to the config of the system tenant. This commit exposes that
access and adjusts those tests.

Release note: None
